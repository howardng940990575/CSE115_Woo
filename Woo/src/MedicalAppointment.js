import React, { Component, ReactNode } from 'react';
import { StyleSheet, Text, View, Button, TextInput, Image, Animated, Dimensions, TouchableHighlight, YellowBox, ScrollView } from 'react-native';
import * as firebase from "firebase";
import { Calendar, Agenda } from 'react-native-calendars';

export default class AppointmentScreen extends Component {

    onPressSchedules = () => {

        this.props.navigation.navigate('Schedule')

    }
    onPressRequests = () => {

        this.props.navigation.navigate('Requests')

    }

    onPressSearch = () => {

      this.props.navigation.navigate('SearchSchedule')

  }

    render() {
        return (
            <View style={styles.container}>
                <Button
                    title='Schedule'
                    style = {styles.requests} 
                    onPress={this.onPressSchedules}/>
                <Button
                    title='requests' 
                    style = {styles.requests} 
                    onPress={this.onPressRequests}/>
                <Button
                    title='search'
                    style = {styles.requests} 
                    onPress={this.onPressSearch}/>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        textAlign: 'center',
        justifyContent: 'center',
        alignItems: 'center',
        paddingLeft: 40,
        paddingRight: 40,
        marginTop: 10,
        height: 200
    },
    requests: {
        // textAlign: 'center',
        // justifyContent: 'center',
        // alignItems: 'center',
        // paddingLeft: 40,
        // paddingRight: 40,
        // marginTop: 10,
        // height: 40
    },
});

/*

*/