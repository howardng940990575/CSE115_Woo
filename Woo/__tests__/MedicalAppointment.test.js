import 'react-native';
import React from 'react';
import MedicalAppointment from '../src/MedicalAppointment';
import renderer from 'react-test-renderer';

test('MedicalAppointment SnapShot', ()=> {
    const snap = renderer.create(<MedicalAppointment />).toJSON();
    expect(snap).toMatchSnapshot();
});

/*test('MedicalAppointment query functionality'), ()=>{
    
}*/