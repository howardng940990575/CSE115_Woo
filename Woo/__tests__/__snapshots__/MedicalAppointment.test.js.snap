// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedicalAppointment SnapShot 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
      false,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#ffffff",
        "justifyContent": "center",
        "marginTop": 50,
        "paddingLeft": 40,
        "paddingRight": 40,
        "textAlign": "center",
      },
    ]
  }
>
  <RCTScrollView>
    <View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
              false,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "borderRadius": 10,
              },
              Object {
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 10,
              },
              undefined,
              undefined,
              undefined,
              Object {
                "alignSelf": "center",
                "width": "85%",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 20,
                "fontWeight": "bold",
                "marginTop": 20,
              }
            }
          >
            Full Appointment
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../assets/appointment_list.png",
              }
            }
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "alignSelf": "center",
                "height": 230,
                "marginBottom": 20,
                "width": 230,
              }
            }
          />
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
              false,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "borderRadius": 10,
              },
              Object {
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 10,
              },
              undefined,
              undefined,
              undefined,
              Object {
                "alignSelf": "center",
                "width": "85%",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 20,
                "fontWeight": "bold",
                "marginTop": 30,
              }
            }
          >
            Search Appointment
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../assets/appointment_search.png",
              }
            }
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "alignSelf": "center",
                "height": 230,
                "marginBottom": 20,
                "width": 230,
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
